tests/app/configure-app-spec.test.jsx

```diff
@@ -0,0 +1,365 @@
+      links: [],
+      extensions: undefined,
+    };
+
+    it('does not create a link with only path', () => {
+      siteConfig.navConfig.navigation.links = [{
+        path: '/site/mock',
+      }];
+
+      testRoutesAndNavigation(siteConfig, mockBasicCC, expectedRoutes, expectedNavigation);
+    });
+
+    it('does not create a link with only a name', () => {
+      siteConfig.navConfig.navigation.links = [{
+        text: 'Mock',
+      }];
+
+      testRoutesAndNavigation(siteConfig, mockBasicCC, expectedRoutes, expectedNavigation);
+    });
+
+    it('does not create a link with only an exampleType', () => {
+      siteConfig.navConfig.navigation.links = [{
+        exampleType: 'mock',
+      }];
+
+      testRoutesAndNavigation(siteConfig, mockBasicCC, expectedRoutes, expectedNavigation);
+    });
+
+    it('does not create a link without an exampleType', () => {
+      siteConfig.navConfig.navigation.links = [{
+        path: '/site/mock',
+        text: 'Mock',
+      }];
+
+      testRoutesAndNavigation(siteConfig, mockBasicCC, expectedRoutes, expectedNavigation);
+    });
+
+    describe('does not create a link without an name', () => {
+      siteConfig.navConfig.navigation.links = [{
+        path: '/site/mock',
+        exampleType: 'mock',
+      }];
+
+      testRoutesAndNavigation(siteConfig, mockBasicCC, expectedRoutes, expectedNavigation);
+    });
+
+    describe('does not create a link without an path', () => {
+      siteConfig.navConfig.navigation.links = [{
+        text: 'Mock',
+        exampleType: 'mock',
+      }];
+
+      testRoutesAndNavigation(siteConfig, mockBasicCC, expectedRoutes, expectedNavigation);
+    });
+
+    Object.keys(componentConfigs).forEach((configKey) => {
+      const componentConfig = componentConfigs[configKey];
+      if (configKey !== 'pagesAndTests') {
+        it(`creates a ${configKey} link with the default component`, () => {
+          siteConfig.navConfig.navigation.links = [{
+            path: '/site/mock',
+            text: 'Mock',
+            exampleType: configKey,
+          }];
+
+          if (configKey === 'tests') {
+            expectedRoutes = ['/site/mock', '/raw/tests'];
+            expectedNavigation.links = [];
+          } else {
+            expectedRoutes = ['/site/mock'];
+            expectedNavigation.links = [{
+              path: '/site/mock',
+              text: 'Mock',
+            }];
+          }
+
+          testRoutesAndNavigation(siteConfig, componentConfig, expectedRoutes, expectedNavigation);
+        });
+
+        it(`creates a ${configKey} link with a custom component`, () => {
+          siteConfig.navConfig.navigation.links = [{
+            path: '/site/mock',
+            text: 'Mock',
+            exampleType: configKey,
+            component: MockComponent,
+          }];
+
+          testRoutesAndNavigation(siteConfig, componentConfig, expectedRoutes, expectedNavigation);
+          // TEST THE CUSTOM COMPONENT
```

Comment:
Within the context, I have yet to find a good way to test this. The may need to be tested within the Index-spec file to fully mound this page. A little tricky as it gets passed as a function at this point.